internal/controller/eventing/eventmesh.go

```diff
@@ -0,0 +1,185 @@
+package eventing
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"github.com/kyma-project/eventing-manager/api/v1alpha1"
+	"github.com/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"os"
+
+	"github.com/kyma-project/kyma/components/eventing-controller/pkg/deployment"
+	"github.com/kyma-project/kyma/components/eventing-controller/pkg/object"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+func (r *Reconciler) startEventMeshSubscriptionController(ctx context.Context, eventing *v1alpha1.Eventing) error {
+	// retrieve secret to authenticate with EventMesh
+	eventMeshSecret, err := r.kubeClient.GetSecret(ctx, eventing.GetEventMeshBackend().Config.EventMeshSecret)
+	if err != nil {
+		return err
+	}
+	// CreateOrUpdate deployment for publisher proxy secret
+	secretForPublisher, err := r.SyncPublisherProxySecret(ctx, eventMeshSecret)
+	if err != nil {
+		// backendStatus.SetPublisherReadyCondition(false, eventingv1alpha1.ConditionReasonPublisherProxySecretError, err.Error())
+		// if updateErr := r.syncBackendStatus(ctx, backendStatus, nil); updateErr != nil {
+		// return ctrl.Result{}, fmt.Errorf("failed to update status while syncing Event Publisher secret: %v", err)
+		// }
+		return err
+	}
+
+	// Set environment with secrets for BEB subscription controller
+	err = setUpEnvironmentForEventMesh(secretForPublisher)
+	if err != nil {
+		// backendStatus.SetSubscriptionControllerReadyCondition(false, eventingv1alpha1.ConditionReasonControllerStartFailed, err.Error())
+		// if updateErr := r.syncBackendStatus(ctx, backendStatus, nil); updateErr != nil {
+		// 	return ctrl.Result{},fmt.Errorf(err, "failed to update status while setting up environment variables for BEB controller")
+		// }
+		return fmt.Errorf("failed to setup environment variables for EventMesh controller: %v", err)
+	}
+
+	// TODO: start subscription controller
```

Comment:
Do we still need this TODO? Is there a follow up ticket for it? Please link it here if any.